Add ability for mapping floats to reversed ranges

ZevEisenberg opened the pull request:
@heyltsjay check it out. Not sure if it's worth it, but this was fun to put together.

heyltsjay commented:
@ZevEisenberg I think my vote may just be for a defaulted boolean `inverted`. I bet when people consume this, they will figure out their ranges through math / prints / debugging, and will plug them in and go. They'll then see that the relationship is backwards and will just want to flip something. I think it's more important to indicate that the relationship is inverted than implying that the directionality of either argument is important. (both descending == both ascending, either one descending == same result)

```titleView.offset = scrollView.contentOffset.y.scaled(from: 100...150, to: 0...30, inverted: true, clamped: true)```

ZevEisenberg commented:
@heyltsjay great point about the direction of both arguments. With my API, there are four permutations of direction (↑↑, ↑↓, ↓↑, ↓↓), but only two distinct outcomes (normal, reversed). We should almost definitely switch to your version.

jvisenti commented:
@ZevEisenberg If the map function were to take a `reversed` `Bool`, then there would be no need for a `ReversedClosedRange` right?

ZevEisenberg commented:
@jvisenti right. It significantly reduces the API surface. Then, just reverse the upper and lower of one of either the "from" or "to" ranges. I'll do that and push it to this branch.

ZevEisenberg commented:
Updated. I also updated the parameter name from `clamp` to `clamped` to match the `clamped(to:)` API that it calls down to, and updated the doc comment. How's this?

```swift
/// Re-maps a number from one range to another.
///
/// - Parameters:
///   - source: The range to interpret the number as being a part of.
///   - destination: The range to map the number to.
///   - clamped: Whether the result should be clamped to the `to` range. Defaults to `false`.
///   - reversed: whether the output mapping should be revserd, such that
///               as the intput increases, the output decreases. Defaults to `false`.
/// - Returns: The input number, scaled from the `from` range to the `to` range.
```

jvisenti commented:
Cleaner, but one downside is that you can't "reverse" the source range, only the destination. I don't think that matters much because you could just switch the params

heyltsjay commented:
@jvisenti per the discussion the above, it doesn't matter which range is switched. there are only two possible mapping outcomes.

heyltsjay commented on Pod/Classes/Math/FloatingPoint+Scale.swift:
```diff
@@ -13,18 +13,21 @@ public extension FloatingPoint {
-    ///   - clamp: Whether the result should be clamped to the `to` range.
+    ///   - clamped: Whether the result should be clamped to the `to` range. Defaults to `false`.
+    ///   - reversed: whether the output mapping should be revserd, such that
+    ///               as the intput increases, the output decreases. Defaults to `false`.
```
typo: `intput`
